pkg/controller/workload/kubernetes/application/application.go

```diff
@@ -310,7 +291,9 @@ func (r *Reconciler) Reconcile(req reconcile.Request) (reconcile.Result, error)
 	}
 
 	if app.DeletionTimestamp != nil {
-		return r.local.delete(ctx, app), errors.Wrapf(r.kube.Update(ctx, app), "cannot update %s %s", v1alpha1.KubernetesApplicationKind, req.NamespacedName)
+		// There's nothing for us to do - the garbage collector will delete any
```

Comment:
I don't think we even need this section. If the object is garbage collected, this condition would be trapped in block `if kerrors.IsNotFound(err) {` above.

Reply:
Good call. Will remove this.